pkg/schema/schema.go

```diff
@@ -252,9 +252,12 @@ func inferTypeFromValue(value interface{}, position *filepos.Position) (yamlmeta
 		return &ScalarType{ValueType: *new(bool), defaultValue: typedContent, Position: position}, nil
 	case nil:
 		return nil, nil
+	case yamlmeta.Node:
+		return nil, fmt.Errorf("Expected value '%s' to be a map, array, or scalar, but was %s", value, typedContent.DisplayName())
+	default:
+		return nil, fmt.Errorf("Expected value '%s' to be a map, array, or scalar, but was %T", value, value)
 	}
 
```

Comment:
```suggestion
```
Style comment: newline can be removed

Reply:
Thanks for the review. Committed the changes.